api/configuration/v1alpha1/konnect_controlplaneref_types.go

```diff
@@ -19,7 +19,7 @@ type ControlPlaneRef struct {
 	// - konnectID
 	// - konnectNamespacedRef
 	// +kubebuilder:validation:Enum=konnectID;konnectNamespacedRef
-	Type string `json:"type,omitempty"`
+	Type string `json:"type"`
```

Comment:
Do we want to make this change now? I believe we want to stick with this being optional to allow implicit defaulting to KIC ( which does not have its own type's `const` defined yet).

Reply:
This makes `Type` required when `ControlPlaneRef` is not nil, which is the right way I think therefore I wanted to fix this straight away.

Reply:
ACK, 👍